src/vsg/io/FileSystem.cpp

```diff
@@ -278,3 +279,47 @@ bool vsg::makeDirectory(const Path& path)
 
     return true;
 }
+
+Path vsg::executableFilePath(void)
+{
+    Path path;
+    char buf[1024];
+
+#if defined(WIN32)
+    DWORD result = GetModuleFileName(NULL, buf, sizeof(buf)-1);
+    if (result && result < sizeof(buf))
+        path = buf;
+#else
+    ssize_t len = readlink("/proc/self/exe", buf, sizeof(buf)-1);
+    if (len != -1)
+        path = buf;
+#endif
+    return path;
+}
+
+Path vsg::executablePath(void)
+{
+    return vsg::filePath(executableFilePath());
+}
+
+Path vsg::sharedDataPath()
+{
+    Path path = executablePath();
+    Path sharePath = path + std::string(1, delimiterNative) + std::string("share");
```

Comment:
Shuld be possible to replace with concatPaths(path, "share");

Reply:
It only works using `concatPaths(path, std::string("share"));`